File: Javascript/NumbersToLetters.js
// Level: 7kyu
// Name: Numbers to Letters
// Instructions: Given an array of numbers, you must return a string. 
// The numbers correspond to the letters of the alphabet in reverse order: a = 26, z = 1 etc. 
// You should also account for '!', '?' and ' ' that are represented by '27', '28' and '29' respectively.

// Solution 1 in Javascript:
function switcher(x){
    const alphabet = "zyxwvutsrqponmlkjihgfedcba!? ";
        return x.map(l => alphabet[l - 1]).join('')
    }

// Solution 2:
function switcher(x){
    const obj = {1:'z',2:'y',3:'x',4:'w',5:'v',6:'u',7:'t',8:'s',9:'r',10:'q',
  11:'p',12:'o',13:'n',14:'m',15:'l',16:'k',17:'j',18:'i',19:'h',20:'g',
  21:'f',22:'e',23:'d',24:'c',25:'b',26:'a',27:'!',28:'?',29:' '}
    return x.map(v=>obj[v*1]).join('')
  }

// Sample Tests Passed:
// Test.describe("Example tests",_=>{
// Test.assertEquals(switcher(['24', '12', '23', '22', '4', '26', '9', '8']), 'codewars');
// Test.assertEquals(switcher(['25','7','8','4','14','23','8','25','23','29','16','16','4']), 'btswmdsbd kkw'); 
// Test.assertEquals(switcher(['4', '24']), 'wc'); 
// });

File: C#/NumbersToLetters.cs
// Language: C#
// Level: 7kyu
// Name of Problem: Numbers to Letters

// Instructions: Given an array of numbers, you must return a string. 
// The numbers correspond to the letters of the alphabet in reverse order: a = 26, z = 1 etc. 
// You should also account for '!', '?' and ' ' that are represented by '27', '28' and '29' respectively.

// Solution:
using System;
using System.Linq;

public class Kata
{
    public static string Switcher(string[] x)
  {
    return string.Join("", x.Select(alpha => "zyxwvutsrqponmlkjihgfedcba!? "[int.Parse(alpha) - 1]).ToArray());  
  }
}

// Sample Tests Passed:
// namespace Solution
// {
//     using NUnit.Framework;
    
//     using System;
    
//     [TestFixture]
//     public class Tests
//     {
//         [Test]
//         public void ExampleTest1()
//         {
//             string expected = "codewars"; 
            
//             string actual = Kata.Switcher(new string[] { "24", "12", "23", "22", "4", "26", "9", "8" });
        
//             Assert.AreEqual(expected, actual);
//         }
        
//         [Test]
//         public void ExampleTest2()
//         {
//             string expected = "btswmdsbd kkw"; 
            
//             string actual = Kata.Switcher(new string[] { "25", "7", "8", "4", "14", "23", "8", "25", "23", "29", "16", "16", "4" });
        
//             Assert.AreEqual(expected, actual);
//         }
        
//         [Test]
//         public void ExampleTest3()
//         {
//             string expected = "wc"; 
            
//             string actual = Kata.Switcher(new string[] { "4", "24" });
        
//             Assert.AreEqual(expected, actual);
//         }
//     }
// }